http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/config/Config.scala

```diff
@@ -458,17 +463,85 @@ object Config {
       dataprocConfig.regionName
     )
   }
-
   val gceMonitorConfig = config.as[GceMonitorConfig]("gce.monitor")
   val dataprocMonitorConfig = config.as[DataprocMonitorConfig]("dataproc.monitor")
   val uiConfig = config.as[ClusterUIConfig]("ui")
-  val serviceAccountProviderClass = config.as[String]("serviceAccounts.providerClass")
   val samAuthConfig = config.as[SamAuthProviderConfig]("auth.providerConfig")
   val httpSamDap2Config = config.as[HttpSamDaoConfig]("auth.providerConfig")
   val liquibaseConfig = config.as[LiquibaseConfig]("liquibase")
   val welderConfig = config.as[WelderConfig]("welder")
   val dbConcurrency = config.as[Long]("mysql.concurrency")
 
+  implicit val kubeClusterConfigReader: ValueReader[KubernetesClusterConfig] = ValueReader.relative { config =>
+    KubernetesClusterConfig(config.as[Location]("location"))
+  }
+
+  implicit val dummyNodepoolConfig: ValueReader[DummyNodepoolConfig] = ValueReader.relative { config =>
+    DummyNodepoolConfig(
+      config.as[MachineTypeName]("machineType"),
+      config.as[NumNodes]("numNodes"),
+      config.as[Boolean]("autoscalingEnabled")
+    )
+  }
+
+  implicit val galaxyNodepoolConfig: ValueReader[GalaxyNodepoolConfig] = ValueReader.relative { config =>
+    GalaxyNodepoolConfig(
+      config.as[MachineTypeName]("machineType"),
+      config.as[NumNodes]("numNodes"),
+      config.as[Boolean]("autoscalingEnabled"),
+      config.as[AutoscalingConfig]("autoscalingConfig")
+    )
+  }
+
+  implicit val autoscalingConfig: ValueReader[AutoscalingConfig] = ValueReader.relative { config =>
+    AutoscalingConfig(
+      config.as[AutoscalingMin]("autoscalingMin"),
+      config.as[AutoscalingMax]("autoscalingMax")
+    )
+  }
+
+  implicit val appConfigReader: ValueReader[GalaxyAppConfig] = ValueReader.relative { config =>
+    GalaxyAppConfig(
+      config.as[ReleaseName]("releaseName"),
+      config.as[NamespaceName]("namespaceNameSuffix"),
+      config.as[List[ServiceConfig]]("services")
+    )
+  }
+
+  implicit val releaseNameReader: ValueReader[ReleaseName] = stringValueReader.map(ReleaseName)
+  implicit val namespaceNameReader: ValueReader[NamespaceName] = stringValueReader.map(NamespaceName)
+
+  implicit val serviceReader: ValueReader[ServiceConfig] = ValueReader.relative { config =>
+    ServiceConfig(
+      config.as[ServiceName]("name"),
+      config.as[KubernetesServiceKindName]("kind"),
+      List() //TODO fill this out if we need ports
```

Comment:
In my initial testing, I haven't had to specify a port number when I construct the URL with the service name. However, a service can have multiple ports, so we may need to specify this when we get further down the implementation path for more complex use cases